object_manager.go

```diff
@@ -615,28 +615,28 @@ func (objMgr *ObjectManager) GetTXTRecord(name string) (*RecordTXT, error) {
 	return &res[0], nil
 }
 
-func (objMgr *ObjectManager) UpdateTXTRecord(recordname string, text string) (*RecordTXT, error) {
+func (objMgr *ObjectManager) UpdateTXTRecord(recordname string, text string) (string, error) {
 	var res []RecordTXT
 
 	recordTXT := NewRecordTXT(RecordTXT{Name: recordname})
 
 	err := objMgr.connector.GetObject(recordTXT, "", &res)
 
 	if len(res) == 0 {
-		return nil, nil
+		return "", nil
 	}
 
 	res[0].Text = text
 
 	res[0].Zone = "" //  set the Zone value to "" as its a non writable field
 
-	_, err = objMgr.connector.UpdateObject(&res[0], res[0].Ref)
+	ref, err := objMgr.connector.UpdateObject(&res[0], res[0].Ref)
 
 	if err != nil || res == nil || len(res) == 0 {
-		return nil, err
+		return "", err
 	}
 
-	return &res[0], nil
+	return ref, nil
```

Comment:
I would suggest returning of pointer to the object by updating the reference of the object beforehand.